src/server/pps/server/master.go

```diff
@@ -223,6 +223,18 @@ func (a *apiServer) deletePipelineResources(ctx context.Context, pipelineName st
 			}
 		}
 	}
+	// delete any secrets associated with the pipeline
+	secrets, err := kubeClient.CoreV1().Secrets(a.namespace).List(metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return errors.Wrapf(err, "could not list secrets")
+	}
```

Comment:
I think you can pass the selector directly to the call to `Delete` at least it works from the command line to do `kubectl delete -l foo=bar`.

Reply:
the `DeleteOptions` struct doesn't take a selector, so i think it must be doing a list call behind the scenes